docs/tutorials/acme/rfc2136-provider.rst

```diff
@@ -0,0 +1,171 @@
+is a solid breakdown of how to use the tool.
+
+To get started, we’ll simply run ``nsupdate -k <keyID>`` where keyID is
+the value returned from ``dnssec-keygen``. This will read the key from
+disk and provide a command prompt to issue commands. In general, we want
+to write a simple TXT RR and make sure we can delete it.
+
+::
+
+   $ nsupdate -k <keyID>
+   > update add www1.example.com txt testing
+   > send
+   > … test here with ``nslookup``
+   > update delete www1.example.com txt
+   > send
+   > … test here with ``nslookup``
+
+Any failures to write, read or delete the record will mean that
+cert-manager will not be able to do so either, no matter how well it is
+configured.
+
+Configuration Step 2 - Set up cert-manager
+------------------------------------------
+
+Now we get to the fun stuff, seeing everything work. Remember that we
+need to set up the ACME DNS-01 issuer and challenge mechanism as well as
+the ``rfc2136`` provider. Since the documentation covers the other parts
+sufficiently, let’s focus on the provider here.
+
+.. code:: yaml
+       rfc2136:
+         nameserver: <address of authoritative nameserver configured above>
+         tsigKeyName: <key name used in `dnssec-keygen`, use something semantically meaningful in both environments>
+         tsigAlgorithm: HMACSHA512 // should be matched to the algo you chose in `dnssec-keygen`
+         tsigSecretSecretRef:
+           name: <the name of the k8s secret holding the TSIG key.. not the key itself!>
+           namespace: <define if not `default`>
+           key: <name of the key *inside* the secret>
+
+Example:
+
+.. code:: yaml
+
+       rfc2136:
+         nameserver: 1.2.3.4:53
+         tsigKeyName: example-com-secret
+         tsigAlgorithm: HMACSHA512
+         tsigSecretSecretRef:
+           name: tsig-secret
+           namespace: cert-manager
+           key: tsig-secret-key
+
+For this example configuration, we’ll need the following two commands.
+The first, on your ``named`` server generates the key. Note how
+``example-com-secret`` is both in the ``tsigKeyName`` above and the
+``dnssec-keygen`` command that follows.
+
+::
+
+   dnssec-keygen -r /dev/urandom -a HMAC-SHA512 -b 512 -n HOST example-com-secret
+
+Also note how the ``tsigAlgorithm`` is provided in both the
+configuration and the keygen command. They are listed at
+https://github.com/miekg/dns/blob/v1.0.12/tsig.go#L18-L23.
+
+The second bit of configuration you need on the kubernetes side is to
+create a secret. Pulling the secret key string from the
+``<key>.private`` file generated above, use the secret in the
+placeholder below:
+
+::
+
+   kubectl -n cert-manager create secret generic tsig-secret --from-literal=tsig-secret-key=<somesecret>
+
+Note how the ``tsig-secret`` and ``tsig-secret-key`` match the
+configuration in the ``tsigSecretSecretRef`` above.
+
+What’s next?
+------------
+
+This configuration so far will actually do nothing. You still have to
+request a certificate!
```

Comment:
Can we link off to the ACME DNS01 tutorial/user guide here, so the user has a place to go 😄

Reply:
Yes, that was the goal, I don't know RST, this code was all generated by a markdown-to-RST site, lol...